com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs

```diff
@@ -14,91 +14,128 @@
         {
-            UrdfInertial urdfInertial = linkObject.AddComponent<UrdfInertial>();
-
-            ArticulationBody robotLink = urdfInertial.GetComponent<ArticulationBody>();
-
-            if (inertial != null)
+            var inertialUrdf = linkObject.AddComponent<UrdfInertial>();
+            if (inertialLink != null)
             {
-                robotLink.mass = ((float)inertial.mass > 0)?((float)inertial.mass):minMass;
-                if (inertial.origin != null) {
-                    
-                    robotLink.centerOfMass = UrdfOrigin.GetPositionFromUrdf(inertial.origin);
-                }
-                else
-                {
-                    robotLink.centerOfMass = Vector3.zero;
-                }
-                urdfInertial.ImportInertiaData(inertial);
-                 
-                urdfInertial.useUrdfData = true;
+                inertialUrdf.m_OriginalValues = inertialLink;
+                inertialUrdf.useUrdfData = true;
+            }
+            else if (inertialUrdf.TryGetComponent<ArticulationBody>(out var robotLink))
+            {
+                //robotLink.inertiaTensor = FixMinInertia(robotLink.inertiaTensor);
+                robotLink.mass = Mathf.Max(robotLink.mass, k_MinMass);
+
+                inertialUrdf.m_Overrides = inertialUrdf.ToLinkInertial(robotLink);
+                // NOTE: The first time this is set to true, we'll save the current state of m_Overrides as the default,
+                //       since there is no actual URDF data to default to
+                inertialUrdf.useUrdfData = false;
             }
+            inertialUrdf.UpdateLinkData();
+        }
 
-            urdfInertial.displayInertiaGizmo = false;
+        public void ResetInertial()
+        {
+            m_Overrides = m_OriginalValues;
+            AssignUrdfInertiaData(m_Overrides);
         }
 
 #region Runtime
 
-        private void Start()
+        void Start()
         {
             UpdateLinkData();
         }
 
-        public void UpdateLinkData()
+        public void UpdateLinkData(bool copyOverrides = false)
         {
-
-            ArticulationBody robotLink = GetComponent<ArticulationBody>();
-
+            var articulationBody = GetComponent<ArticulationBody>();
+            if (articulationBody == null)
+            {
+                return;
+            }
+            
             if (useUrdfData)
             {
-                robotLink.centerOfMass = centerOfMass;
-                robotLink.inertiaTensor = inertiaTensor;
-                robotLink.inertiaTensorRotation = inertiaTensorRotation * inertialAxisRotation;
+                if (m_OriginalValues == null)
+                {
+                    Debug.LogWarning(
+                        "This instance doesn't have any urdf data stored - " +
+                        "creating some using the current inertial values.");
+                    m_OriginalValues = ToLinkInertial(articulationBody);
```

Comment:
Do we want to create fake "original" values here? Maybe makes more sense to output an error and keep original data as is?

Reply:
The original values aren't faked here - they are inferred by the current values on the `ArticulationBody` because the `URDFInertial` doesn't store any references to the original URDF file from which it read its parameters. I think it's safe to assume that the `ArticulationBody` values correspond 1:1 with the original URDF values, since before the change in this PR they would have been immutable (hence the "bug"), but I'm not totally sure...